docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+between the ``minval`` and ``maxval`` for each hyperparameter for each trial.
+
+..
+   note:
+
+   To see early stopping in action, try setting ``max_trials`` to over 500 and playing around with the hyperparamter search values.
+
+In this step, we’ll run our experiment using the model_def_adaptive.py script and its accompanying
+adaptive.yaml experiment configuration file.
+
+Begin by accessing the hyperparameters in your code:
+
+..
+   note:
+
+   Refer to line 204 in ``model_def_adaptive.py``
+
+.. code::
+
+   #NEW: Get hparams chosen for this trial from cluster info object.
+   hparams = info.trial.hparams
+
+.. image:: /assets/images/model-def-adaptive-script.png
+
+Then, pass the hyperparameters into your model and optimizer:
+
+.. code::
+
+   #NEW: Pass relevant hparams to model and optimizer.
+   model = Net(hparams).to(device) optimizer =
+   optim.Adadelta(model.parameters(), lr=hparams[“learning_rate”])
+
+Ensure your model is set to use the selected values on a per-trial basis rather than your previously
+hardcoded values:
+
+..
+   note:
+
+   Refer to line 18 in ``model_def_adaptive.py``
+
+.. code::
+
+   class Net(nn.Module):
+       # NEW: Add hparams to **init**.
+       def **init**\ (self, hparams):
+           # NEW: Read hyperparameters provided for this trial.
+           super(Net, self).__init()
+           self.conv1 = nn.Conv2d(1,hparams[“n_filters1”], 3, 1)
+           self.conv2 = nn.Conv2d(hparams[“n_filters1”], hparams[“n_filters2”], 3, 1)
+           self.dropout1 = nn.Dropout(hparams[“dropout1”])
+           self.dropout2 = nn.Dropout(hparams[“dropout2”])
+           self.fc1 = nn.Linear(144 * hparams[“n_filters2”], 128)
+           self.fc2 = nn.Linear(128, 10)
+
+Step 4.1: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code::
+
+   det e create adaptive.yaml .
+
+In the Determined WebUI, navigate to the **Hyperparameters** tab.
+
+You should see a graph in the WebUI that displays the various trials initiated by the Adaptive ASHA
+hyperparameter search algorithm.
+
+******************************
+ Step 5: Distributed Training
+******************************
+
+The Core API has special considerations for running distributed training. Some of the more important
+considerations are:
+
+-  Access to all IP addresses of every node in the Trial (through the ClusterInfo API).
+
+-  Communication primitives such as :meth:`~determined.core.DistributedContext.allgather`,
+   :meth:`~determined.core.DistributedContext.gather`, and
+   :meth:`~determined.core.DistributedContext.broadcast` to give you out-of-the-box coordination
+   between workers.
+
+-  Since many distributed training frameworks expect all workers in training to operate in-step, the
+   :meth:`~determined.core.PreemptContext.should_preempt` call is automatically synchronized across
+   workers so that all workers decide to preempt or continue as a unit.
+
+..
+   tip Launchers
```

Comment:
again, sphinx is reading this as a commment, not a tip directive

Reply:
resolved